File: GeneratedEntityReaderClasses/Components/GameLogComponent.cs
namespace GeneratedNoitaClasses;
public class GameLogComponent : NoitaComponentBase,  iNoitaType<GameLogComponent>{
	public NoitaBool Deleted { get; set; } = new();
	public NoitaBool Enabled { get; set; } = new();
	public NoitaString Tags { get; set; } = new();
	public NoitaBool report_death { get; set; } = new();
	public NoitaBool report_damage { get; set; } = new();
	public NoitaBool report_new_biomes { get; set; } = new();
	public NoitaVector<NoitaString> mVisitiedBiomes { get; set; } = new();

    public void Read(NoitaStream s){
		Deleted.Read(s);
		Enabled.Read(s);
		Tags.Read(s);
		report_death.Read(s);
		report_damage.Read(s);
		report_new_biomes.Read(s);
		mVisitiedBiomes.Read(s);

    }
    public void Write(NoitaStream s){
		Deleted.Write(s);
		Enabled.Write(s);
		Tags.Write(s);
		report_death.Write(s);
		report_damage.Write(s);
		report_new_biomes.Write(s);
		mVisitiedBiomes.Write(s);

    }
}

File: GeneratedEntityReaderClassesPython/GeneratedClasses/Components/GameLogComponent.py
from dataclasses import dataclass, field
from BasicNoitaTypes import NoitaBool
from BasicNoitaTypes import NoitaString
from BasicNoitaTypes import NoitaVector
from BasicNoitaTypes import NoitaString

from NoitaStream import NoitaStream
@dataclass
class GameLogComponent:
    Deleted:NoitaBool = field(default_factory=NoitaBool)
    Enabled:NoitaBool = field(default_factory=NoitaBool)
    Tags:NoitaString = field(default_factory=NoitaString)
    report_death:NoitaBool = field(default_factory=NoitaBool)
    report_damage:NoitaBool = field(default_factory=NoitaBool)
    report_new_biomes:NoitaBool = field(default_factory=NoitaBool)
    mVisitiedBiomes:NoitaVector = field(default_factory=lambda: NoitaVector(NoitaString))
    
    def Read(self, s:NoitaStream):
        self.Deleted.Read(s)
        self.Enabled.Read(s)
        self.Tags.Read(s)
        self.report_death.Read(s)
        self.report_damage.Read(s)
        self.report_new_biomes.Read(s)
        self.mVisitiedBiomes.Read(s)
    
    def Write(self, s:NoitaStream):
        self.Deleted.Write(s)
        self.Enabled.Write(s)
        self.Tags.Write(s)
        self.report_death.Write(s)
        self.report_damage.Write(s)
        self.report_new_biomes.Write(s)
        self.mVisitiedBiomes.Write(s)

